base/applications/mspaint/winproc.cpp

```diff
@@ -280,14 +280,18 @@ LRESULT CMainWindow::OnInitMenuPopup(UINT nMsg, WPARAM wParam, LPARAM lParam, BO
     BOOL trueSelection =
         (::IsWindowVisible(selectionWindow) &&
          ((toolsModel.GetActiveTool() == TOOL_FREESEL) || (toolsModel.GetActiveTool() == TOOL_RECTSEL)));
+    BOOL isBMP;
     switch (lParam)
     {
         case 0: /* File menu */
             if ((HMENU)wParam != GetSubMenu(menu, 0))
                 break;
-            EnableMenuItem(menu, IDM_FILEASWALLPAPERPLANE,     ENABLED_IF(isAFile));
-            EnableMenuItem(menu, IDM_FILEASWALLPAPERCENTERED,  ENABLED_IF(isAFile));
-            EnableMenuItem(menu, IDM_FILEASWALLPAPERSTRETCHED, ENABLED_IF(isAFile));
+
+            isBMP = _wcsicmp(PathFindExtensionW(filepathname), L".bmp") == 0;
```

Comment:
I think there should be a way to try setting the wallpaper, and check error code in case it fails.

Reply:
I think the current way of making paint check that itself is better (more defensive) compared to passing garbage data to other components and relying on their correct behavior for all of them.
It is more likely that we will have to adapt the code again in the future to support other formats (like jpg), yes. But unless you could provide demo code for the check you have in mind, I do like the current approach.

I still recommend to double-check the beahvior in our unpatched build for .jpg and .png before merging this PR.

Reply:
For the record: On my XPSP3 the desk.cpl applet allows to set as wallpaper *.bmp, *.gif, *.jpg, *.jpeg, *.dib, *.png
And it calls that set of formats "Alle Bilddateien".
We do know that different Windows versions do support a different set of formats. Most likely they get more with each version of Windows.

Furthermore my XPSP3 allows *.htm and *.html but since we decided to not support ActiveDesktop that part doesn't apply to us. And I think this is a good decision. Active Desktop sucks.

Reply:
I investigated that and our desk.cpl is actually able to set more formats as wallpapers (e.g. gif, png, jpg) than our Paint. So it seems that Paint does something wrong when setting images as wallpapers, so it only works with bmp.
As for the list of formats, our desk.cpl calls gdi to find supported formats, but this approach also isn't correct (for example, it also counts .ico, but you can't set an icon as a wallpaper in desk.cpl).